### PR TITLE
fix: multi filters for one field

### PR DESCRIPTION
**Pull request by Yuqi-Du**
Bug:
If a query run with multiple filter for the same field it's not resolved correctly.
Query {"find":{"filter":{"product_price":{"$ne":9.99, "$exists" : true}}}}, only one of the condition is resolved to filter. 

Solved by adding additional ComparisonExpression.

**Which issue(s) this PR fixes**:
Fixes[ #<763>](https://github.com/stargate/jsonapi/issues/763)

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by maheshrajamani — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java [hunk 1]
```diff
@@ -69,6 +69,16 @@ public void addComparisonExpression(ComparisonExpression comparisonExpression) {
     comparisonExpressions.add(comparisonExpression);
   }
 
+  public void addComparisonExpressions(List<ComparisonExpression> comparisonExpressionList) {
+    for (ComparisonExpression comparisonExpression : comparisonExpressionList) {
+      if (comparisonExpression.getPath().equals(DocumentConstants.Fields.DOC_ID)) {
+        totalIdComparisonExpressionCount++;
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java:75]: Can you add documentation for these counts?

File: src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java [hunk 2]
```diff
@@ -550,6 +550,53 @@ ValueComparisonOperator.NE, new JsonLiteral(value, JsonType.SUB_DOC))),
           .isEqualTo(expectedResult.getPath());
     }
 
+    @Test
+    public void mustHandleMultiFilterWithOneField() throws Exception {
+      String json =
+          """
+                {
+                   "name": {
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java:558]: Would be great to have an integration test like this as well

**Review by vkarpov15 — COMMENTED** [review]
Minor suggestion otherwise LGTM

**Pull request merged**